src/Raven.Server/Documents/Queries/Results/QueryResultRetrieverBase.cs

```diff
@@ -1162,7 +1260,8 @@ private static bool TryGetTimeSeriesFunction(string methodName, Query query, out
                    func.Type == DeclaredFunction.FunctionType.TimeSeries;
         }
 
-        private bool TryGetFieldValueFromDocument(Document document, FieldsToFetch.FieldToFetch field, out object value)
+        private bool TryGetFieldValueFromDocument<TDocument>(TDocument document, FieldsToFetch.FieldToFetch field, out object value)
```

Comment:
Please check the warning - `Type parameter 'TDocument' has the same name as the type parameter from outer type 'QueryResultRetrieverBase<TDocument>'`